src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt

```diff
@@ -33,9 +37,16 @@ class VisitNotificationEventService(
 
   @Transactional
   fun handleNonAssociations(notificationDto: NonAssociationChangedNotificationDto) {
-    if (prisonerService.hasPrisonerGotANonAssociationWith(notificationDto.prisonerNumber, notificationDto.nonAssociationPrisonerNumber)) {
-      val affectedVisits = getOverLappingVisits(notificationDto)
-      processVisitsWithNotifications(affectedVisits, NON_ASSOCIATION_EVENT)
+    with(notificationDto) {
+      if (NON_ASSOCIATION_CREATED == type) {
+        val affectedVisits = getOverLappingVisits(notificationDto)
+        processVisitsWithNotifications(affectedVisits, NON_ASSOCIATION_EVENT)
+      } else if (type in arrayOf(NON_ASSOCIATION_DELETED, NON_ASSOCIATION_CLOSED)) {
+        if (!prisonerService.hasPrisonerGotANonAssociationWith(prisonerNumber, nonAssociationPrisonerNumber)) {
```

Comment:
is the ! (not) incorrect?

Reply:
Yes, basically if there is no non associations it means we can safely delete, other wise there could muliple notifications for mulitile non associations for the same visits and we don't know which one!